msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplication.kt

```diff
@@ -111,5 +188,375 @@ class NativeAuthPublicClientApplication(
+                            error = result.error,
+                            correlationId = result.correlationId,
+                            errorCodes = result.errorCodes
+                        )
+                    )
+                }
+                is SignInCommandResult.PasswordRequired -> {
+                    SignInResult.PasswordRequired(
+                        nextState = SignInPasswordRequiredState(
+                            flowToken = result.credentialToken,
+                            scopes = scopes,
+                            config = nativeAuthConfig
+                        )
+                    )
+                }
+                is SignInCommandResult.InvalidCredentials -> {
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result $result"
+                    )
+                    SignInResult.UnexpectedError(
+                        error = GeneralError(
+                            errorMessage = "Unexpected state",
+                            error = "unexpected_state",
+                            correlationId = result.correlationId,
+                            errorCodes = result.errorCodes
+                        )
+                    )
+                }
+                is SignInCommandResult.Complete -> {
+                    Logger.warn(
+                        TAG,
+                        "Unexpected result $result"
+                    )
+                    SignInResult.UnexpectedError(
+                        error = GeneralError(
+                            errorMessage = "Unexpected state",
+                            error = "unexpected_state",
+                            correlationId = "UNSET"
+                        )
+                    )
+                }
+                is ICommandResult.Redirect -> {
+                    SignInResult.BrowserRequired(
+                        error = BrowserRequiredError(
+                            correlationId = result.correlationId
+                        )
+                    )
+                }
+            }
+        }
+    }
+
+    interface SignInUsingPasswordCallback : Callback<SignInUsingPasswordResult>
+
+    /**
+     * Sign in the account using username and password; callback variant.
+     *
+     * @param username username of the account to sign in.
+     * @param password password of the account to sign in.
+     * @param scopes (Optional) list of scopes to request.
+     * @param callback [com.microsoft.identity.client.NativeAuthPublicClientApplication.SignInUsingPasswordCallback] to receive the result.
+     * @return [com.microsoft.identity.client.statemachine.results.SignInUsingPasswordResult] see detailed possible return state under the object.
+     * @throws MsalClientException if an account is already signed in.
+     */
+    override fun signInUsingPassword(
+        username: String,
+        password: String,
```

Comment:
We must use char[] instead of String, for security concerns. https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords

Reply:
Pleaes apply this to everywhere the password is used in this PR, then overwrite the data as soon as we no longer need the password.

Reply:
I have changed string to char array for password.